wasm/derive/src/entrypoint.rs

```diff
@@ -3,96 +3,31 @@
 #![allow(clippy::str_to_string)]
 
 use proc_macro::TokenStream;
+use proc_macro_error::abort;
 use quote::quote;
-use syn::{parse_macro_input, parse_quote, punctuated::Punctuated};
-
-mod kw {
-    pub mod param_types {
-        syn::custom_keyword!(authority);
-        syn::custom_keyword!(triggering_event);
-    }
-}
-
-/// Enum representing possible attributes for [`entrypoint`] macro
-enum Attr {
-    /// List of parameters
-    Params(super::params::ParamsAttr<ParamType>),
-    /// Empty attribute. Used when attribute input is empty
-    Empty,
-}
-
-impl syn::parse::Parse for Attr {
-    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
-        if input.is_empty() {
-            return Ok(Attr::Empty);
-        }
-
-        Ok(Attr::Params(input.parse()?))
-    }
-}
-
-/// Type of smart contract entrypoint function parameter.
-///
-/// *Type* here means not just *Rust* type but also a purpose of a parameter.
-/// So that it uses [`Authority`](ParamType::Authority) instead of `account::Id`.
-enum ParamType {
-    Authority,
-    TriggeringEvent,
-}
-
-impl syn::parse::Parse for ParamType {
-    fn parse(input: syn::parse::ParseStream) -> syn::parse::Result<Self> {
-        use kw::param_types::*;
-
-        super::parse_keywords!(input,
-            authority => ParamType::Authority,
-            triggering_event => ParamType::TriggeringEvent,
-        )
-    }
-}
-
-impl super::params::ConstructArg for ParamType {
-    fn construct_arg(&self) -> syn::Expr {
-        match self {
-            ParamType::Authority => {
-                parse_quote! {
-                    ::iroha_wasm::query_authority()
-                }
-            }
-            ParamType::TriggeringEvent => {
-                parse_quote! {{
-                    use ::iroha_wasm::debug::DebugExpectExt as _;
-
-                    let top_event = ::iroha_wasm::query_triggering_event();
-                    ::iroha_wasm::debug::DebugExpectExt::dbg_expect(
-                        ::core::convert::TryInto::try_into(top_event),
-                        "Failed to convert top-level event to the concrete one"
-                    )
-                }}
-            }
-        }
-    }
-}
+use syn::{parse_macro_input, parse_quote};
 
 /// [`entrypoint`](crate::entrypoint()) macro implementation
-pub fn impl_entrypoint(attr: TokenStream, item: TokenStream) -> TokenStream {
+pub fn impl_entrypoint(item: TokenStream) -> TokenStream {
     let syn::ItemFn {
         attrs,
         vis,
         sig,
         mut block,
     } = parse_macro_input!(item);
 
-    assert!(
-        syn::ReturnType::Default == sig.output,
-        "Exported function must not have a return type"
-    );
+    if syn::ReturnType::Default != sig.output {
```

Comment:
Yoda notation?

Reply:
let me speak like yoda

Reply:
Jokes aside, this is useful in C-like languages.